src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -0,0 +1,229 @@
+                }
+            }
+
+            if constexpr (sizeof...(S) != 0)
+            {
+                Validate<S...>(root, settings, warnings);
+            }
+        }
+    }
+
+    namespace details
+    {
+        std::optional<SettingMapping<Setting::AutoUpdateTimeInMinutes>::value_t>
+        SettingMapping<Setting::AutoUpdateTimeInMinutes>::Validate(SettingMapping<Setting::AutoUpdateTimeInMinutes>::json_t value)
+        {
+            return value;
+        }
+
+        std::optional<SettingMapping<Setting::ProgressBarVisualStyle>::value_t>
+        SettingMapping<Setting::ProgressBarVisualStyle>::Validate(SettingMapping<Setting::ProgressBarVisualStyle>::json_t value)
+        {
+            // progressBar property possible values
+            static constexpr std::string_view s_progressBar_Accent = "accent";
+            static constexpr std::string_view s_progressBar_Rainbow = "rainbow";
+            static constexpr std::string_view s_progressBar_Retro = "retro";
+
+            if (Utility::CaseInsensitiveEquals(value, s_progressBar_Accent))
+            {
+                return VisualStyle::Accent;
+            }
+            else if (Utility::CaseInsensitiveEquals(value, s_progressBar_Rainbow))
+            {
+                return VisualStyle::Rainbow;
+            }
+            else if (Utility::CaseInsensitiveEquals(value, s_progressBar_Retro))
+            {
+                return VisualStyle::Retro;
+            }
+
+            return {};
+        }
+    }
+
+    UserSettings::UserSettings() : m_type(UserSettingsType::Default)
+    {
+        Json::Value settingsRoot = Json::Value::nullSingleton();
+
+        // Settings can be loaded from settings.json or settings.json.backup files.
+        // 1 - Use settings.json if exists and passes parsing.
+        // 2 - Use settings.backup.json if settings.json fails to parse.
+        // 3 - Use default (empty) if both settings files fail to load.
+
+        auto settingsJson = ParseFile(s_SettingFileName, m_warnings);
+        if (settingsJson.has_value())
+        {
+            m_type = UserSettingsType::Standard;
+            settingsRoot = settingsJson.value();
+        }
+
+        // Settings didn't parse or doesn't exist, try with backup.
+        if (settingsRoot.isNull())
+        {
+            auto settingsBackupJson = ParseFile(s_SettingBackupFileName, m_warnings);
+            if (settingsBackupJson.has_value())
+            {
+                m_warnings.emplace_back(SettingsWarnings::LoadedBackupSettings);
+                m_type = UserSettingsType::Backup;
+                settingsRoot = settingsBackupJson.value();
+            }
+        }
+
+        if (!settingsRoot.isNull())
+        {
+            Validate<
+                Setting::AutoUpdateTimeInMinutes,
+                Setting::ProgressBarVisualStyle
+                >(settingsRoot, m_settings, m_warnings);
```

Comment:
Should probably just make Validate take a single template param and have multiple lines.